File: typescript/allLongestStrings.ts
// Given an array of strings, return another array containing all of its longest strings.

// Example

// For inputArray = ["aba", "aa", "ad", "vcd", "aba"], the output should be
// solution(inputArray) = ["aba", "vcd", "aba"].

function solution(inputArray: string[]): string[] {
  let maxSize: number = Math.max(
    ...inputArray.map((currentString) => currentString.length)
  );
  return inputArray.filter((currentString) => currentString.length === maxSize);
}

File: python/all_longest_strings.py
# Given an array of strings, return another array containing all of its longest strings.

# Example

# For inputArray = ["aba", "aa", "ad", "vcd", "aba"], the output should be
# solution(inputArray) = ["aba", "vcd", "aba"].

def solution(inputArray):
    maxSize = max(map(lambda currentString: len(currentString), inputArray))
    longest = filter(lambda currentString: len(currentString) == maxSize, inputArray)
    return list(longest)
    